components/eamxx/src/diagnostics/vertical_layer_interface.cpp

```diff
@@ -7,9 +7,16 @@ namespace scream
 VerticalLayerInterfaceDiagnostic::VerticalLayerInterfaceDiagnostic (const ekat::Comm& comm, const ekat::ParameterList& params)
```

Comment:
Do you think it's worth it to also merge the int and mid version into a single `VerticalLayer` diagnostic? There seem to be a lot of code duplication, and the only difference is an extra call to `calculate_z_mid` in the mid case (something that we could handle with a simple if statement)...

Reply:
Yeah, a lot of duplicated code. I'll add that.